src/serac/physics/mesh.cpp

```diff
@@ -41,9 +43,15 @@ Mesh::Mesh(mfem::ParMesh&& mesh, const std::string& meshtag) : mesh_tag_(meshtag
   meshtmp->EnsureNodes();
   meshtmp->ExchangeFaceNbrData();
   mfem_mesh_ = &serac::StateManager::setMesh(std::move(meshtmp), mesh_tag_);
+  errorIfRankHasNoElements();
   createDomains();
 }
 
+Mesh::errorIfRankHasNoElements() const
+{
+  SLIC_ERROR_IF(mfem_mesh_->GetNE() == 0, "Local size of mesh is 0 which will cause out-of-range error");
```

Comment:
```suggestion
  SLIC_ERROR_IF(mfem_mesh_->GetNE() == 0, "After refining and distributing mesh, local size of mesh is 0");
```

not sure the last part is helpful and may cause confusion.